Simplified/NYPLSettingsAccountDetailViewController.m

```diff
@@ -538,26 +538,26 @@ - (void)tableView:(__attribute__((unused)) UITableView *)tableView
   
   switch(cellKind) {
     case CellKindAgeCheck: {
+      [self.tableView deselectRowAtIndexPath:indexPath animated:YES];
       UITableViewCell *cell = [tableView cellForRowAtIndexPath:indexPath];
-      if (self.selectedAccount.details.userAboveAgeLimit == YES) {
-        [self confirmAgeChange:^(BOOL under13) {
-          if (under13) {
-            cell.accessoryView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"CheckboxOff"]];
-            self.selectedAccount.details.userAboveAgeLimit = NO;
-            //Delete Books in My Books
-            [[NYPLMyBooksDownloadCenter sharedDownloadCenter] reset:self.selectedAccountId];
-            [[NYPLBookRegistry sharedRegistry] reset:self.selectedAccountId];
+      
+      if (!NYPLSettings.shared.userPresentedAgeCheck) {
+        __weak NYPLSettingsAccountDetailViewController *weakSelf = self;
+        [[[AccountsManager shared] ageCheck] verifyCurrentAccountAgeRequirementWithUserAccountProvider:self.businessLogic.userAccount
```

Comment:
The user could be signed into another library (eg. NYPL) and try to perform age check on SimplyE.
Using the account within `businessLogic` here handles such edge case.